enhancements/installer/aws-cross-account-dns-zone.md

```diff
@@ -0,0 +1,294 @@
+
+
+```go
+	// platform holds configuration specific to the underlying
+	// infrastructure provider for DNS.
+	// When omitted, this means the user has no opinion and the platform is left
+	// to choose reasonable defaults. These defaults are subject to change over time.
+	//
+	// +optional
+	Platform DNSPlatformSpec `json:"platform,omitempty"`
+```
+
+```go
+// DNSPlatformSpec holds cloud-provider-specific configuration
+// for DNS administration.
+// +union
+type DNSPlatformSpec struct {
+	// type is the underlying infrastructure provider for the cluster.
+	// Allowed values are "AWS", "Azure", "BareMetal", "GCP", "Libvirt",
+	// "OpenStack", "VSphere", "oVirt", "KubeVirt", "EquinixMetal", "PowerVS",
+	// "AlibabaCloud", "Nutanix" and "None". Individual components may not support all platforms,
+	// and must handle unrecognized platforms as None if they do not support that platform.
+	//
+	// +unionDiscriminator
+	Type PlatformType `json:"type"`
+
+	// aws contains DNS configuration specific to the Amazon Web Services cloud provider.
+	// +optional
+	AWS *AWSDNSSpec `json:"aws,omitempty"`
+}
+
+// AWSDNSSpec contains DNS configuration specific to the Amazon Web Services cloud provider.
+type AWSDNSSpec struct {
+	// privateZoneIAMRole contains the ARN of a role that should be assumed when performing
+	// operations on the cluster's private hosted zone specified in the cluster DNS config.
+	// When left empty, no role should be assumed.
+	// +kubebuilder:validation:Pattern:=`^arn:(aws|aws-cn|aws-us-gov):iam:[a-z0-9-]+:[0-9]{12}:role\/.*$`
+	// +optional
+	PrivateZoneIAMRole string `json:"privateZoneIAMRole,omitempty"`
+}
+
+```
+
+The name of the field is discussed further in open questions.
+
+#### Install Config
+
+`platform.aws.hostedZoneRole` would be added to complement the existing `platform.aws.hostedZone` field.
+
+An example of `platform` from `install-config.yaml`, specifying subnets of the shared VPC,
+a hosted zone ID, and a role to assume:
+
+```yaml
+platform:
+  aws:
+    region: us-east-2
+    subnets:
+    - subnet-0ef71407239a73558
+    - subnet-00e576c91f8578661
+    - subnet-041af6e1bbf29c2ef
+    - subnet-037ab7dc0cd396721
+    - subnet-0adda52a1007fc74f
+    - subnet-09bd88678851acab6
+    hostedZone: Z0743278N5ELJPLYA3H6
+    hostedZoneRole: arn:aws:iam::<account-a>:role/<role-name>
+```
+Please see Open Questions for further discussion of the install config.
+
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+ With the implementation of this API, the installer and cluster components should follow a
+ relatively straightforward pattern: if the role field is not empty,
+ call [`AssumeRole`](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/stscreds/)
+ when performing operations on the private hosted zone.
+
+### Risks and Mitigations
+
+n/a
```

Comment:
Add-on operators needing permissions will not work unless updated to use the role ARN.

Reply:
@mrunalp are you describing the external-dns-operator here? I think that's the only operator that is affected by this feature.

Reply:
@gcs278 Yes, that's what I was thinking about.